internal/service/amp/workspace_data_source.go

```diff
@@ -0,0 +1,79 @@
+package amp
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/prometheusservice"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+)
+
+func DataSourceWorkspace() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceWorkspaceRead,
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"workspace_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"prometheus_endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"alias": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tags": tftags.TagsSchemaComputed(),
+		},
+	}
+}
+
+func dataSourceWorkspaceRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).AMPConn
+	ignoreTagsConfig := meta.(*conns.AWSClient).IgnoreTagsConfig
+
+	workspaceID := d.Get("workspace_id").(string)
+	workspace, err := FindWorkspaceByID(conn, workspaceID)
+
+	if err != nil {
+		return fmt.Errorf("error reading AMP Workspace (%s): %w", workspaceID, err)
+	}
+
+	d.SetId(workspaceID)
+
+	workspaceARN := arn.ARN{
```

Comment:
Is the arn building by hand necessary? https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-APIReference.html#AMP-APIReference-DescribeWorkspace returns the arn generated by the service code

Reply:
I was curious as well but every data source I've looked at has a similar convention. Here is an example from ebs_volume_data_source.go

```
	arn := arn.ARN{
		Partition: meta.(*conns.AWSClient).Partition,
		Service:   ec2.ServiceName,
		Region:    meta.(*conns.AWSClient).Region,
		AccountID: meta.(*conns.AWSClient).AccountID,
		Resource:  fmt.Sprintf("volume/%s", d.Id()),
	}
	d.Set("arn", arn.String())
```
Seems Arns are treated differently than other describe fields.

Reply:
to my best knowledge, the construction of ARNs by hand is only required for services that do not return an ARN in their API response. 
Turns out there are a lot of AWS services that don't return an ARN, and the ARNs are required ie for the AWS resource tagging api.

\cc @jeromeinsf

Reply:
Just made a commit to leverage describe workspace for arn just like the other fields.

```
	d.Set("arn", workspace.Arn)
	d.Set("prometheus_endpoint", workspace.PrometheusEndpoint)
```